src/plugins/github/fetchGithubRepo.js

```diff
@@ -163,47 +162,56 @@ function tryGithubFetch(fetch, fetchOptions): Promise<any> {
             x.errors[0].type === "RATE_LIMITED"
           ) {
             return Promise.reject(
-              ({
-                type: "RATE_LIMIT_EXCEEDED",
-                retry: false,
-                error: x,
-              }: GithubResponseError)
+              ({type: "RATE_LIMIT_EXCEEDED", error: x}: GithubResponseError)
             );
           } else {
             return Promise.reject(
-              ({
-                type: "GRAPHQL_ERROR",
-                retry: false,
-                error: x,
-              }: GithubResponseError)
+              ({type: "GRAPHQL_ERROR", error: x}: GithubResponseError)
             );
           }
         }
         if (x.data === undefined) {
           if (x.message && x.message.includes("Bad credentials")) {
             return Promise.reject(
-              ({
-                type: "BAD_CREDENTIALS",
-                retry: false,
-                error: x,
-              }: GithubResponseError)
+              ({type: "BAD_CREDENTIALS", error: x}: GithubResponseError)
             );
           } else {
             // See https://github.com/sourcecred/sourcecred/issues/350
             return Promise.reject(
-              ({type: "NO_DATA", retry: true, error: x}: GithubResponseError)
+              ({type: "NO_DATA", error: x}: GithubResponseError)
             );
           }
         }
         return Promise.resolve(x.data);
       }),
     (e) =>
-      Promise.reject(
-        ({type: "FETCH_ERROR", retry: false, error: e}: GithubResponseError)
-      )
+      Promise.reject(({type: "FETCH_ERROR", error: e}: GithubResponseError))
   );
 }
 
+function errorDisposition(
```

Comment:
Now that the error handling logic is separated from the fetch logic, would be good to add tests for this

Reply:
Hmm—unit tests here don’t seem useful to me. The only nontrivial case is
`RATE_LIMIT_EXCEEDED`, so we could test that. But the tests would be
immediately removed in #2327, because this function becomes impure. Note
that in #2327 I _do_ add unit tests for the new logic.

What tests would you like to see, to fix which failure mode?

> Naming nit: `getErrorDisposition` instead of just `errorDisposition`

No; superfluous `get`s provide no information, and we avoid them. See
`Graph.node`, `MarkovProcessGraph.participants`, `CredView.weights`,
`CacheProvider.database`, `Ledger.account`, ….